assemblies/resources/bin/check_ports

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+# Licensed to The Apereo Foundation under one or more contributor license
+# agreements. See the NOTICE file distributed with this work for additional
+# information regarding copyright ownership.
+#
+#
+# The Apereo Foundation licenses this file to you under the Educational
+# Community License, Version 2.0 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of the License
+# at:
+#
+#   http://opensource.org/licenses/ecl2.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# -----------------------------------------------------------------------
+# check if http or https port is available to run opencast otherwise exit
+
+if [ -d "$1" ]; then
+  SETENV_ETC="$1"
+else
+  if [ -d "${KARAF_BASE}" ]; then
+    SETENV_ETC="${KARAF_BASE}/etc"
+  else
+    SETENV_ETC="etc"
+  fi
+fi
+
+check_port() {
```

Comment:
I tried this function on an Ubuntu system and couldn't get it to reliable work:
```
$ uname -a
Linux [...] 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

$ netstat -tulpn
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN      870/systemd-resolve
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      1204/sshd
tcp        0      0 0.0.0.0:6783            0.0.0.0:*               LISTEN      1596/weaver
tcp        0      0 127.0.0.1:6784          0.0.0.0:*               LISTEN      1596/weaver
tcp6       0      0 :::80                   :::*                    LISTEN      1278/dockerd
tcp6       0      0 :::22                   :::*                    LISTEN      1204/sshd
tcp6       0      0 :::443                  :::*                    LISTEN      1278/dockerd
tcp6       0      0 :::6782                 :::*                    LISTEN      1596/weaver
tcp6       0      0 :::2377                 :::*                    LISTEN      1278/dockerd
tcp6       0      0 :::7946                 :::*                    LISTEN      1278/dockerd
udp        0      0 0.0.0.0:4789            0.0.0.0:*                           -
udp        0      0 0.0.0.0:6783            0.0.0.0:*                           1596/weaver
udp        0      0 0.0.0.0:6784            0.0.0.0:*                           -
udp        0      0 127.0.0.53:53           0.0.0.0:*                           870/systemd-resolve
udp6       0      0 :::6784                 :::*                                -
udp6       0      0 :::7946                 :::*                                1278/dockerd
...

$ check_port 22    # -> 0
$ check_port 53    # -> 0
$ check_port 68    # -> 0
$ check_port 80    # -> 0
$ check_port 443   # -> 0
$ check_port 546   # -> 0
$ check_port 2377  # -> 0

$ check_port 4789  # -> 1
$ check_port 6782  # -> 1
$ check_port 6783  # -> 1
$ check_port 6784  # -> 1
$ check_port 7946  # -> 1
```

Reply:
Thanks ... proc stores the port numbers as 4 digit Hex, I'll fix this

Reply:
This is working correctly now.